include/wx/osx/core/colour.h

```diff
@@ -29,60 +29,78 @@ class WXDLLIMPEXP_CORE wxColour: public wxColourBase
     // default copy ctor and dtor are ok
 
     // accessors
-    virtual bool IsOk() const { return m_cgColour != NULL; }
+    virtual bool IsOk() const wxOVERRIDE;
 
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Red() const { return m_red; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Green() const { return m_green; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Blue() const { return m_blue; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Alpha() const { return m_alpha; }
+    virtual ChannelType Red() const wxOVERRIDE;
+    virtual ChannelType Green() const wxOVERRIDE;
+    virtual ChannelType Blue() const wxOVERRIDE;
+    virtual ChannelType Alpha() const wxOVERRIDE;
+
+    wxColour& operator=(const wxColour& col);
 
     // comparison
     bool operator == (const wxColour& colour) const;
-
     bool operator != (const wxColour& colour) const { return !(*this == colour); }
 
-    CGColorRef GetPixel() const { return m_cgColour; }
+    // CoreGraphics CGColor
+    // --------------------
 
-    CGColorRef GetCGColor() const { return m_cgColour; }
-    CGColorRef CreateCGColor() const { return wxCFRetain( (CGColorRef)m_cgColour ); }
-
-#if wxOSX_USE_COCOA_OR_CARBON
-    void GetRGBColor( RGBColor *col ) const;
-#endif
-
-    // Mac-specific ctor and assignment operator from the native colour
-    // assumes ownership of CGColorRef
+    // This ctor does take ownership of the color.
     wxColour( CGColorRef col );
+    
+    // don't take ownership of the returned value
+    CGColorRef GetCGColor() const;
+    
+    // do take ownership of the returned value
+    CGColorRef CreateCGColor() const { return wxCFRetain(GetCGColor()); }
+
 #if wxOSX_USE_COCOA_OR_CARBON
+    // Quickdraw RGBColor
+    // ------------------
     wxColour(const RGBColor& col);
-    wxColour& operator=(const RGBColor& col);
+   void GetRGBColor( RGBColor *col ) const;
 #endif
+
 #if wxOSX_USE_COCOA
+    // NSColor Cocoa
+    // -------------
+
     // This ctor does not take ownership of the color.
     explicit wxColour(WX_NSColor color);
     WX_NSColor OSXGetNSColor() const;
 #endif
-    wxColour& operator=(CGColorRef col);
-    wxColour& operator=(const wxColour& col);
-
+ 
 protected :
     virtual void
-    InitRGBA(ChannelType r, ChannelType g, ChannelType b, ChannelType a);
-#if wxOSX_USE_COCOA_OR_CARBON
-    void InitRGBColor( const RGBColor& col );
-#endif
-    void InitCGColorRef( CGColorRef col );
+    InitRGBA(ChannelType r, ChannelType g, ChannelType b, ChannelType a) wxOVERRIDE;
 
-private:
-    wxCFRef<CGColorRef>     m_cgColour;
+    virtual wxGDIRefData *CreateGDIRefData() const wxOVERRIDE;
+    virtual wxGDIRefData *CloneGDIRefData(const wxGDIRefData *data) const wxOVERRIDE;
 
-    ChannelType             m_red;
-    ChannelType             m_blue;
-    ChannelType             m_green;
-    ChannelType             m_alpha;
+private:
 
     wxDECLARE_DYNAMIC_CLASS(wxColour);
 };
 
+class wxColorRefData : public wxGDIRefData
```

Comment:
Also, this should really be called `wxColourRefData`, with "u", for consistency.

Reply:
yes, thanks